pkg/config/parameter/file/file.go

```diff
@@ -0,0 +1,78 @@
+/*
+ * @license
+ * Copyright 2024 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package file
+
+import (
+	"fmt"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/strings"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/template"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/config/parameter"
+	"github.com/spf13/afero"
+	"path/filepath"
+)
+
+const FileParameterType = "file"
+
+var FileParameterSerde = parameter.ParameterSerDe{
+	Serializer:   writeFileValueParameter,
+	Deserializer: parseFileValueParameter,
+}
+
+type FileParameter struct {
+	WorkingDir string
+	Folder     string
+	Path       string
+}
+
+func (f *FileParameter) GetType() string {
+	return FileParameterType
+}
+
+func (f *FileParameter) GetReferences() []parameter.ParameterReference {
+	// file parameters cannot have references
+	return []parameter.ParameterReference{}
+}
+
+func (f *FileParameter) ResolveValue(context parameter.ResolveContext) (interface{}, error) {
+	content, err := afero.ReadFile(afero.NewOsFs(), filepath.Join(f.WorkingDir, f.Folder, f.Path))
```

Comment:
The direct use of the OS filesystem is the only part of this change that I'm unsure about. As nothing is actively writing this to files, it's not an issue that we can't inject the CopyOnWrite filesystem that we usually use in testing - but generally, it makes this parameter directly tied to the OS Fs rather the Fs that's actually being used by the config loader

It may be overkill - but I think we could hand in the same Fs that is used by the config-loader via the ResolveContext?

Reply:
in a previous implementation i used a BasePath Fs. It required passing around that fs five times on the caller side just to support that special parameter loading. This and the fact that it is still 100% testable, and the fact that this type of parameter is -by definition- tied to the file system made me decide to do it like that.

Reply:
also, `ResolveContext` currently does not make any assumptions on "how" parameters are loaded, which is nice.

Reply:
I can live with it in general, especially because it's not an issue in production use (currently) 
But it still feels a like a potentially hard to debug case if something goes wrong. 

Thinking of an E2E test case: the copy-on-write FS is used and config files are rewritten. Now this parameter might read something else than expected by a test as the test would write to an overlay FS (in memory or temp files) but the parameter will always read from the OS Fs. 
The fact that this parameter uses than the rest of our loading code feels potentially brittle

Reply:
@UnseenWizzard I've pushed another [commit](https://github.com/Dynatrace/dynatrace-configuration-as-code/pull/1461/commits/994c55f95f4fb4051d3b11f3c4eb5976c07ee911) as an alternative implementation (reusing the afero filesystem used for config loading). :) please take a look and lets see if it is better :) also @arthurpitman  pls take a look 👋

Reply:
nice side effect, because of https://github.com/Dynatrace/dynatrace-configuration-as-code/pull/1461/files#diff-66ed2aae4cea3078ecd8fe5a748f3402a57747f3eb02981d07038496ade82ff5R133 we don't need to store working dir, nor folder into the parameter anymore 👌